components/Animate/AnimateInView.tsx

```diff
@@ -1,4 +1,5 @@
-import React, { useRef } from 'react';
+'use client';
```

Comment:
All components that involves interaction need the "use client" directive. Components are server components by default, but for interaction to happen, we need them to be client components.

Also, I removed the src/ directory so components, hooks, utilities directory are all in the root directory now. This follows the most conventional project structure (for the app/ approach) as recommended by nextjs